tests/buildkite/build-cuda.sh

```diff
@@ -22,7 +22,7 @@ command_wrapper="tests/ci_build/ci_build.sh gpu_build_centos7 docker --build-arg
 echo "--- Build libxgboost from the source"
 $command_wrapper tests/ci_build/prune_libnccl.sh
 $command_wrapper tests/ci_build/build_via_cmake.sh -DUSE_CUDA=ON -DUSE_NCCL=ON \
-  -DUSE_OPENMP=ON -DHIDE_CXX_SYMBOLS=ON -DUSE_NCCL_LIB_PATH=ON \
+  -DUSE_OPENMP=ON -DPLUGIN_FEDERATED=ON -DHIDE_CXX_SYMBOLS=ON -DUSE_NCCL_LIB_PATH=ON \
```

Comment:
Is gRPC a runtime dependency? If so, we shouldn't enable the federated learning plugin in the Python wheel on PyPI.

Reply:
We can built a separate artifact with gRPC for testing purposes